### DISCUSSION TITLE
Inserting data in 2 tables together

### DISCUSSION THREAD

**Discussion by chahal26** [filamentphp/filament#3036] [Help] [answered]
Hi, I have 2 tables i.e. **users and warehouses**. There are a few fields in the user's table that need to be inserted and the rest in warehouses. 

We have a `user_id` field in the warehouse's table using which we are having a `User::hasOne(Warehouse::class)` relationship.

I want to create a resource for the warehouse where all details will be asked together. But when create is clicked, **first data is inserted in the user's table and then, its user_id is inserted in the warehouse's table with the rest of the data**. 

Can someone please guide me on whats the best way to achieve this?

**Comment by ManojKiranA** [comment, marked as answer]
you can use relationship to do this.

https://filamentphp.com/docs/2.x/forms/layout#saving-data-to-relationships

**Reply by chahal26** [reply]
I tried but that is working when we have a relationship row already created i.e. user is already created. I need to create both together.

**Reply by ManojKiranA** [reply]
![image](https://user-images.githubusercontent.com/30294553/177575741-ee16e55b-ab4b-40b0-bc98-1ac720bcf10a.png)
